### ISSUE TITLE
Postpone population of the NodeSettings or derived feature settings to the FullNodeBuilder.Build method

### ISSUE THREAD

**Issue by quantumagi** [stratisproject/StratisBitcoinFullNode#1051]
We will only know the exact feature list of a node once the Build method is called which makes it more convenient to read / process the configuration settings and command line arguments at that time. After this change it would be easier to:
- print command line help which only relates to the relevant features or 
- creating a default configuration file that only includes settings for the relevant features.

**Comment by fassadlr** [comment]
@quantumagi Interesting idea! I just had a look quick and I see that what we get the node settings in the Build method already?

[FullNodeBuilder](https://github.com/stratisproject/StratisBitcoinFullNode/blob/master/src/Stratis.Bitcoin/Builder/FullNodeBuilder.cs#L166)

Sorry, can you maybe elaborate a bit more on what you are proposing?

**Comment by quantumagi** [comment]
We basically just need to check that none of the node settings are being used prior. Also the PrintHelp method is currently being used "too early" - before we know which features to print help for. I suspect the creation of a default configuration files is also happening too early.

**Comment by dangershony** [comment]
I agree we could improgve this however that means certain parts of feature will have to use settings only in the feature initialize method
